rust/kernel/types.rs

```diff
@@ -316,6 +316,18 @@ impl<T> Opaque<T> {
     pub fn get(&self) -> *mut T {
         UnsafeCell::raw_get(self.0.as_ptr())
     }
+
+    /// Gets the value behind `this`.
+    ///
+    /// This function is useful to get access to the value without creating intermeditate
+    /// references.
+    ///
+    /// # Safety
+    ///
+    /// The pointer supplied is valid.
+    pub unsafe fn raw_get(this: *const Self) -> *mut T {
+        UnsafeCell::raw_get(addr_of!((*this).0).cast::<UnsafeCell<T>>())
```

Comment:
Seems reasonable, it was a bit unclear to me if we guarantee that `Opaque` will always be `transparent` (I guess changing it would also require changing this method, but wanted to be careful). To be clear: I do not think that there is a reason for making `Opaque` not transparent, so we could just clarify that in the documentation.